spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/ManagedCommitSuite.scala

```diff
@@ -116,6 +123,405 @@ class ManagedCommitSuite
+
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 1
+      Seq(3).toDF.write.format("delta").mode("append").save(tablePath) // version 2
+      DeltaLog.clearCache()
+      commitStore.numGetCommitsCalled = 0
+      import testImplicits._
+      val result1 = sql(s"SELECT * FROM delta.`$tablePath`").collect()
+      assert(result1.length === 2 && result1.toSet === Set(Row(2), Row(3)))
+      assert(commitStore.numGetCommitsCalled === 2)
+    }
+  }
+
+  test("cold snapshot initialization - with and without initial commit store") {
+    CommitStoreProvider.registerBuilder(TrackingInMemoryCommitStoreBuilder(batchSize = 10))
+    withTempDir { tempDir =>
+      val tablePath = tempDir.getAbsolutePath
+      Seq(1).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 0
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 1
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 2
+      DeltaLog.clearCache()
+
+      // Case-1: Create deltaLog with initial commit store
+      val usageRecords1 = Log4jUsageLogger.track {
+        val log = DeltaLog.forTable(spark, new Path(tablePath))
+        val snapshot = log.unsafeVolatileSnapshot
+        assert(snapshot.version === 2)
+        assert(snapshot.commitStoreOpt.nonEmpty)
+      }
+      assert(filterUsageRecords(usageRecords1, "delta.readChecksum").size === 2)
+
+      // Case-2: Create deltaLog without initial commit store
+      DeltaLog.clearCache()
+      val usageRecords2 = Log4jUsageLogger.track {
+        val ownerName = spark.conf.get(MANAGED_COMMIT_OWNER_NAME.defaultTablePropertyKey)
+        val ownerConfStr = spark.conf.get(MANAGED_COMMIT_OWNER_CONF.defaultTablePropertyKey)
+        val ownerConf = JsonUtils.fromJson[Map[String, String]](ownerConfStr)
+        val cs = CommitStoreProvider.getCommitStore(ownerName, ownerConf)
+        val log = DeltaLog.forTable(
+          spark,
+          dataPath = new Path(tablePath),
+          options = Map.empty,
+          initialCommitStoreOpt = Some(cs),
+          tableIdentifier = None)
+        val snapshot = log.unsafeVolatileSnapshot
+        assert(snapshot.version === 2)
+        assert(snapshot.commitStoreOpt.nonEmpty)
+      }
+      assert(filterUsageRecords(usageRecords2, "delta.readChecksum").size === 1)
+
+    }
+  }
+
+  // Test commit owner changed on concurrent cluster
+  test("snapshot is updated recursively when FS table is converted to commit owner" +
+      " table on a concurrent cluster") {
+    val commitStore = new TrackingCommitStore(new InMemoryCommitStore(batchSize = 10))
+    val builder = TrackingInMemoryCommitStoreBuilder(batchSize = 10, Some(commitStore))
+    CommitStoreProvider.registerBuilder(builder)
+    val oldCommitOwnerValue = spark.conf.get(MANAGED_COMMIT_OWNER_NAME.defaultTablePropertyKey)
+    spark.conf.unset(MANAGED_COMMIT_OWNER_NAME.defaultTablePropertyKey)
```

Comment:
aside: I guess `withSqlConf` can't handle this `unset` case?

Reply:
Yes exactly.